pkg/operator/sync_openshiftcontrollermanager_v311_00.go

```diff
@@ -158,138 +159,151 @@ func syncOpenShiftControllerManager_v311_00_to_latest(
+	if d.Status.UpdatedReplicas != *d.Spec.Replicas {
+		progressingMessages = append(progressingMessages, fmt.Sprintf(
+			"deployment/%s: updated replicas is %d, desired replicas is %d.",
+			d.Name, d.Status.UpdatedReplicas, *d.Spec.Replicas))
 	}
+
+	// This is actually not depending on the deployment object,
+	// but generation mismatch sets all operands to Progressing.
 	if operatorConfig.ObjectMeta.Generation != operatorConfig.Status.ObservedGeneration {
-		progressingMessages = append(progressingMessages, fmt.Sprintf("openshiftcontrollermanagers.operator.openshift.io/cluster: observed generation is %d, desired generation is %d.", operatorConfig.Status.ObservedGeneration, operatorConfig.ObjectMeta.Generation))
+		progressingMessages = append(progressingMessages, fmt.Sprintf(
+			"openshiftcontrollermanagers.operator.openshift.io/cluster: observed generation is %d, desired generation is %d.",
+			operatorConfig.Status.ObservedGeneration, operatorConfig.ObjectMeta.Generation))
 	}
+
 	if len(progressingMessages) == 0 {
 		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorapiv1.OperatorCondition{
-			Type:   operatorapiv1.OperatorStatusTypeProgressing,
+			Type:   conditionTypePrefix + operatorapiv1.OperatorStatusTypeProgressing,
 			Status: operatorapiv1.ConditionFalse,
 		})
 	} else {
 		v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorapiv1.OperatorCondition{
-			Type:    operatorapiv1.OperatorStatusTypeProgressing,
+			Type:    conditionTypePrefix + operatorapiv1.OperatorStatusTypeProgressing,
 			Status:  operatorapiv1.ConditionTrue,
 			Reason:  "DesiredStateNotYetAchieved",
 			Message: strings.Join(progressingMessages, "\n"),
 		})
 	}
-
-	v1helpers.SetOperatorCondition(&operatorConfig.Status.Conditions, operatorapiv1.OperatorCondition{
-		Type:   operatorapiv1.OperatorStatusTypeUpgradeable,
-		Status: operatorapiv1.ConditionTrue,
-	})
-
-	operatorConfig.Status.ObservedGeneration = operatorConfig.ObjectMeta.Generation
-	resourcemerge.SetDeploymentGeneration(&operatorConfig.Status.Generations, actualDeployment)
-	resourcemerge.SetDeploymentGeneration(&operatorConfig.Status.Generations, actualRCDeployment)
-
-	return syncReturn(c, errors, originalOperatorConfig, operatorConfig)
 }
 
-func syncReturn(c OpenShiftControllerManagerOperator, errors []error, originalOperatorConfig, operatorConfig *operatorapiv1.OpenShiftControllerManager) (bool, error) {
-	if len(errors) > 0 {
-		message := ""
-		for _, err := range errors {
-			message = message + err.Error() + "\n"
+// syncReturn checks the error slices and sets Degraded conditions in the operator config before returning.
+func syncReturn(
+	c OpenShiftControllerManagerOperator,
+	ocmErrors, rcmErrors []error,
+	originalOperatorConfig, operatorConfig *operatorapiv1.OpenShiftControllerManager,
+) (bool, error) {
+	setDegraded := func(errs []error, conditionTypePrefix string) {
+		conditionType := conditionTypePrefix + workloadDegradedCondition
```

Comment:
I am unsure whether we want to keep using `WorkloadDegraded` still as Jira issue mentions `Degraded` only.

Reply:
My mistake, I have updated the Jira issue. Let's keep using WorkloadDegraded to be backwards compatible.

For RCM we can use RouteControllerManagerDegraded IMO.

Reply:
Amended.